pkg/csi/cinder/nodeserver.go

```diff
@@ -286,28 +286,30 @@ func (ns *nodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstag
 }
 
 func (ns *nodeServer) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
-
 	nodeID, err := ns.Metadata.GetInstanceID()
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "[NodeGetInfo] unable to retrieve instance id of node %v", err)
 	}
 
-	zone, err := ns.Metadata.GetAvailabilityZone()
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "[NodeGetInfo] Unable to retrieve availability zone of node %v", err)
+	nodeInfo := &csi.NodeGetInfoResponse{
+		NodeId:            nodeID,
+		MaxVolumesPerNode: ns.Opts.NodeVolumeAttachLimit,
 	}
-	topologyMap := make(map[string]string, len(ns.Topologies)+1)
-	topologyMap[topologyKey] = zone
-	for k, v := range ns.Topologies {
-		topologyMap[k] = v
+
+	if ns.Driver.withTopology {
```

Comment:
Can we have an early return here?

Reply:
Another one taken from the Manila server:

https://github.com/kubernetes/cloud-provider-openstack/blob/dc8e21cdae5657b420f52a972b7d06661dd246f1/pkg/csi/manila/nodeserver.go#L338-L349

Again, do I update that there also?

Reply:
as I mentioned above, let's not mix things up in this PR.

Reply:
are you going to change cinder CSI code here or do this in one show for cinder and manila in one PR?